### DISCUSSION TITLE
Question about the tracker macro...

### DISCUSSION THREAD

**Discussion by edfloreshz** [Relm4/Relm4#132] [Q&A] [answered]
Is there a way to use the `tracker!` macro outside of `#[relm4::widget]`?

For example: 

```rust
impl Widgets<AppModel, ()> for AppWidgets {
    type Root = adw::ApplicationWindow;

    fn init_view(model: &AppModel, _components: &AppComponents, sender: Sender<AppMsg>) -> Self {
        let flap = adw::Flap::builder()
            .reveal_flap(track!(model.changed(AppModel::show_panel()), model.show_panel))
    }
}
```

**Comment by AaronErhardt** [comment, marked as answer]
Outside of the widget macro you could just use the "normal" tracker API:

```rust
if model.changed(AppModel::show_panel()) {
    flap.reveal_flap(model.show_panel);
}
```

https://github.com/AaronErhardt/Tracker

Btw. there's also a shorter syntax for simple tracker expressions: https://github.com/AaronErhardt/Relm4/blob/main/relm4-examples/examples/macro_reference.rs#L130
